types/recharts/index.d.ts

```diff
@@ -535,7 +537,7 @@ export interface RectangleProps extends Partial<CSSStyleDeclaration> {
 
 export class Rectangle extends React.Component<RectangleProps> { }
 
-export interface ReferenceAreaProps {
+export interface ReferenceAreaProps extends CellProps {
```

Comment:
instead of extending ReferenceAreaProps from CellProps, extend them from the cssstyledeclarations...